src/easylink/step.py

```diff
@@ -1643,8 +1795,8 @@ def add_edges_to_implementation_graph(
         # Add the edges at this level (i.e. the edges at this `self._step`)
         for source, target, edge_attrs in self._step.step_graph.edges(data=True):
             edge = EdgeParams.from_graph_edge(source, target, edge_attrs)
-            source_step = self._nodes[source]["step"]
```

Comment:
I decided I preferred the context provided by explicitly calling `self._step.step_graph.nodes` every time